File: video/ai-extensions/video-81-leetcode-9/problem-9.ts

function isPalindrome(x: number): boolean {
  // This is where the AI-generated TypeScript code goes
  if (x < 0) return false;
  let reversed = 0, original = x;
  while (x > 0) {
    reversed = reversed * 10 + (x % 10);
    x = Math.floor(x / 10);
  }
  return original === reversed;
}

File: video/ai-extensions/video-80-leetcode-5/video-81-leetcode-9/problem-9.kt

class Solution {
    fun isPalindrome(x: Int): Boolean {
        if (x < 0 || (x % 10 == 0 && x != 0)) {
            return false
        }
        
        var revertedNumber = 0
        var number = x
        while (number > revertedNumber) {
            revertedNumber = revertedNumber * 10 + number % 10
            number /= 10
        }
        
        return number == revertedNumber || number == revertedNumber / 10
    }
}
